app/uk/gov/hmrc/cdsimportsddsfrontend/services/CustomsDeclarationsService.scala

```diff
@@ -40,18 +40,24 @@ class CustomsDeclarationsService @Inject()(appConfig: AppConfig, declarationXml:
   def submit(eori: Eori, declarationViewModel: DeclarationViewModel)(implicit hc: HeaderCarrier): Future[DeclarationServiceResponse] = {
     val declaration: Declaration = declarationViewModel.toDeclaration
```

Comment:
this isn't your doing, but i think this should be moved out into the controller...  the service shouldn't know about the viewmodel